.github/workflows/release-v2.yml

```diff
@@ -63,48 +75,57 @@ jobs:
            </servers>
           </settings>
           EOF
+
       - name: Build with Maven
         run: |
-          mvn versions:set -DremoveSnapshot versions:commit
-          mvn -P dse -q -ff clean package -DskipTests
-      - name: zip-up
+          ./mvnw versions:set -DremoveSnapshot versions:commit
+          ./mvnw -P dse -q -ff clean package -DskipTests
+
+      - name: Zip-up `stargate-lib`
         run: |
           zip -r stargate-jars.zip starctl* stargate-lib
-      - name: Retrieve stashed release URL
-        uses: actions/download-artifact@v1
-        with:
-          name: release-url
-      - name: Read release URL
-        id: get_release_url
-        run: echo ::set-output name=URL::$(cat release-url/url.txt)
+
+      # uploads the jars by referencing the release-upload-url from create-release job
       - name: Upload jars
         uses: actions/upload-release-asset@v1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          upload_url: ${{ steps.get_release_url.outputs.URL }}
+          upload_url: ${{needs.create-release.outputs.release-upload-url}}
           asset_path: stargate-jars.zip
           asset_name: stargate-jars.zip
-          asset_content_type: text/html
+          asset_content_type: application/zip
 
-  publish:
+  # publish coordinator JARs to the OSSRH
+  publish-ossrh:
+    name: Publish to OSSRH
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+
+      # TODO same here, why do we need the ref?
+      - uses: actions/checkout@v3
         with:
           ref: "v2.0.0"
+
+      # TODO this below is some magic I don't get, why not using ${{ github.ref }} as in create-release
```

Comment:
Would it be easy to test to see if that'd work? The whole "echo" construct is needed for some reason by Github actions to set output variable for other steps to see I think. Sort of idiom although quite incomprehensible to me.